File: Axiom/Internal/Frames/Frame.cs
using Axiom.Internal.Ast;
using System.Collections.Generic;

namespace Axiom.Internal.Frames
{
    internal class Frame
    {
        public Frame(Label label, FunctionExpression functionExpression, int level)
        {
            Function = functionExpression;

            Level = level;
            Label = label;
            Variables = new List<Access>();
            FramePointer = Temp.Create();
            ReturnValue = Temp.Create();
        }

        public override string ToString()
        {
            return string.Format("Frame({0})", Label.Name, Level, Label.Name, Size);
        }

        public int Size { get; set; }

        public FunctionExpression Function { get; private set; }
        public int Level { get; private set; }
        public Label Label { get; private set; }
        public IList<Access> Variables { get; private set; }
        public Temp FramePointer { get; private set; }
        public Temp ReturnValue { get; private set; }
    }
}

File: src/main/java/com/moybl/axiom/frame/Frame.java
package com.moybl.axiom.frame;

import com.moybl.axiom.ast.FunctionExpression;

import java.util.ArrayList;
import java.util.List;

public class Frame {

	private Label label;
	private FunctionExpression functionExpression;
	private int level;
	private int size;
	private List<Access> variables;
	private Temp framePointer;
	private Temp returnValue;

	public Frame(Label label, FunctionExpression functionExpression, int level) {
		this.label = label;
		this.functionExpression = functionExpression;
		this.level = level;

		variables = new ArrayList<>();
		framePointer = Temp.create();
		returnValue = Temp.create();
	}

	public Label getLabel() {
		return label;
	}

	public FunctionExpression getFunctionExpression() {
		return functionExpression;
	}

	public int getLevel() {
		return level;
	}

	public int getSize() {
		return size;
	}

	public List<Access> getVariables() {
		return variables;
	}

	public Temp getFramePointer() {
		return framePointer;
	}

	public Temp getReturnValue() {
		return returnValue;
	}

}
